cmake/pika_add_executable.cmake

```
@@ -209,7 +209,11 @@ function(pika_add_executable name)
   endif()
 
   if(${name}_CUDA)
```

We can probably change the option to `GPU` instead?
```suggestion
  if(${name}_GPU)
```